src/components/event/MapModal/Map-modal.js

```diff
@@ -1,58 +1,54 @@
-﻿import React, { Component } from 'react';
+﻿import React, { useState } from 'react';
 import Button from '@material-ui/core/Button';
 import Dialog from '@material-ui/core/Dialog';
 import DialogActions from '@material-ui/core/DialogActions';
 import DialogContent from '@material-ui/core/DialogContent';
 import DialogTitle from '@material-ui/core/DialogTitle';
-import { LocationMapWithCircle } from '../../helpers/form-helpers/location';
 import { Field } from 'redux-form';
+
+import { LocationMapWithCircle } from '../../helpers/form-helpers/location';
 import '../slider.css';
 import DisplayMap from '../map/display-map';
+import constants from '../../../constants/mapModal';
 
-class MapModal extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            open: false
-        }
-    }
+export const MapModal = ({initialize,values}) => {
+
+    const [open, setOpen] = useState(false);
 
-    handleClickOpen = () => {
-        this.setState({ open: true });
+    const handleClickOpen = () => {
+        setOpen(true);
     };
 
-    handleClose = () => {
-        const startValue = this.props.initialize({
+     const handleClose = () => {
+
+        const startValue = initialize({
             radius: 8,
             selectedPos: { latitude: null, longitude: null }
         })
-        if (this.props.values.selectedPos.latitude != null)
-            return startValue;
-        else
-            return startValue && this.setState({ open: false })  
+
+        !!values.selectedPos.latitude ? startValue :
```

Comment:
There is no need to make extra conversion of this variable to the boolean types. We can just to check if variable have some value: `values.selectedPos.latitude ? ... : ...`

Reply:
Add please destruct `latitude` property somewhere in the component to not refer to this variable with full path(`values.selectedPos.latitude`) each time(I see, that it used a lot of times in this component).